### DISCUSSION TITLE
NPC Bots High Damage

### DISCUSSION THREAD

**Discussion by Kyroth88** [trickerer/Trinity-Bots#695] [General]
I have found while running level 60 bots on vanilla content (eg. vanilla raids, 60 end game dungeons etc.) that bot DPS is extremely high with mediocre gear. 

I am running an Azerothcore 3.3.5a server with the latest build with Autobalance and NPC bots (both of your forked versions). 

My MT bot has DPS around 800-900 (with 1H and shield in dungeon blue gear) and attackers have DPS (again in Dungeon blue gear) fluctuating between 800-1400 DPS depending on AOE.

This seems to be consistent through level 60 content no matter if it is in a raid or dungeon. In a 5 man group in Strat with Autobalance creature stats all at 1.0, the DPS is essentially the same as in a 10 man group raiding Molten Core with Autobalance creature stats around ~0.65.

I am wondering if this is just how the bots scale, or if there is something wrong in my configuration that I should be looking for as I am not completely sure how much impact the gear has on the bots DPS.

For now in the config I have set the level 60-69 bot DPS to 0.5 (down from 1.0) to compensate for this which seems to bring it in line with what I believe the DPS for this level should be ~400-500 DPS.

Thank you very much for this amazing work you have done!!!

**Comment by trickerer** [comment]
I'm not totally aware of AC changes to DPS calculation over last couple months, but there are always at least two things to consider:
- Modules that affect creature damage (like Autobalance)
- https://github.com/trickerer/Trinity-Bots/issues/692#issuecomment-1826177417. When I first merged bots into AC I noticed this damage calculation code being slightly different from what TC has there. Probably a typo I said and added a commented TC version of that code, but never uncommented it. See if that helps

**Reply by Kyroth88** [reply]
Thanks a ton! I will play around with those commented sections and report back. I will also disable AB just to be sure (but based on creature stats and my observations with full groups/adjusted groups I do not think it is the factor here).

**Comment by Kyroth88** [comment]
I can confirm that following this (https://github.com/trickerer/Trinity-Bots/issues/692#issuecomment-1826177417) dramatically helped with bot DPS coming back down to earth. Now I can fine tune it from here. 

Thank you!

**Comment by trickerer** [comment]
https://github.com/trickerer/Trinity-Bots/commit/31efedb71662ac07ab83b212c3f5e8b1179c782c
